arch/xtensa/src/esp32/hardware/esp32_pcnt.h

```diff
@@ -0,0 +1,2495 @@
+#define PCNT_PLUS_CNT_RST_U5_S  10
+
+/* PCNT_CNT_PAUSE_U4 : R/W ;bitpos:[9] ;default: 1'b0 ; */
+
+/* Description: Set this bit to pause unit4's counter. */
+
+#define PCNT_CNT_PAUSE_U4  (BIT(9))
+#define PCNT_CNT_PAUSE_U4_M  (BIT(9))
+#define PCNT_CNT_PAUSE_U4_V  0x1
+#define PCNT_CNT_PAUSE_U4_S  9
+
+/* PCNT_PLUS_CNT_RST_U4 : R/W ;bitpos:[8] ;default: 1'b1 ; */
+
+/* Description: Set this bit to clear unit4's counter. */
+
+#define PCNT_PLUS_CNT_RST_U4  (BIT(8))
+#define PCNT_PLUS_CNT_RST_U4_M  (BIT(8))
+#define PCNT_PLUS_CNT_RST_U4_V  0x1
+#define PCNT_PLUS_CNT_RST_U4_S  8
+
+/* PCNT_CNT_PAUSE_U3 : R/W ;bitpos:[7] ;default: 1'b0 ; */
+
+/* Description: Set this bit to pause unit3's counter. */
+
+#define PCNT_CNT_PAUSE_U3  (BIT(7))
+#define PCNT_CNT_PAUSE_U3_M  (BIT(7))
+#define PCNT_CNT_PAUSE_U3_V  0x1
+#define PCNT_CNT_PAUSE_U3_S  7
+
+/* PCNT_PLUS_CNT_RST_U3 : R/W ;bitpos:[6] ;default: 1'b1 ; */
+
+/* Description: Set this bit to clear unit3's counter. */
+
+#define PCNT_PLUS_CNT_RST_U3  (BIT(6))
+#define PCNT_PLUS_CNT_RST_U3_M  (BIT(6))
+#define PCNT_PLUS_CNT_RST_U3_V  0x1
+#define PCNT_PLUS_CNT_RST_U3_S  6
+
+/* PCNT_CNT_PAUSE_U2 : R/W ;bitpos:[5] ;default: 1'b0 ; */
+
+/* Description: Set this bit to pause unit2's counter. */
+
+#define PCNT_CNT_PAUSE_U2  (BIT(5))
+#define PCNT_CNT_PAUSE_U2_M  (BIT(5))
+#define PCNT_CNT_PAUSE_U2_V  0x1
+#define PCNT_CNT_PAUSE_U2_S  5
+
+/* PCNT_PLUS_CNT_RST_U2 : R/W ;bitpos:[4] ;default: 1'b1 ; */
+
+/* Description: Set this bit to clear unit2's counter. */
+
+#define PCNT_PLUS_CNT_RST_U2  (BIT(4))
+#define PCNT_PLUS_CNT_RST_U2_M  (BIT(4))
+#define PCNT_PLUS_CNT_RST_U2_V  0x1
+#define PCNT_PLUS_CNT_RST_U2_S  4
+
+/* PCNT_CNT_PAUSE_U1 : R/W ;bitpos:[3] ;default: 1'b0 ; */
+
+/* Description: Set this bit to pause unit1's counter. */
+
+#define PCNT_CNT_PAUSE_U1  (BIT(3))
+#define PCNT_CNT_PAUSE_U1_M  (BIT(3))
+#define PCNT_CNT_PAUSE_U1_V  0x1
+#define PCNT_CNT_PAUSE_U1_S  3
+
+/* PCNT_PLUS_CNT_RST_U1 : R/W ;bitpos:[2] ;default: 1'b1 ; */
+
+/* Description: Set this bit to clear unit1's counter. */
+
+#define PCNT_PLUS_CNT_RST_U1  (BIT(2))
+#define PCNT_PLUS_CNT_RST_U1_M  (BIT(2))
+#define PCNT_PLUS_CNT_RST_U1_V  0x1
+#define PCNT_PLUS_CNT_RST_U1_S  2
+
+/* PCNT_CNT_PAUSE_U0 : R/W ;bitpos:[1] ;default: 1'b0 ; */
+
+/* Description: Set this bit to pause unit0's counter. */
+
+#define PCNT_CNT_PAUSE_U0  (BIT(1))
+#define PCNT_CNT_PAUSE_U0_M  (BIT(1))
+#define PCNT_CNT_PAUSE_U0_V  0x1
+#define PCNT_CNT_PAUSE_U0_S  1
+
+/* PCNT_PLUS_CNT_RST_U0 : R/W ;bitpos:[0] ;default: 1'b1 ; */
+
+/* Description: Set this bit to clear unit0's counter. */
+
+#define PCNT_PLUS_CNT_RST_U0  (BIT(0))
+#define PCNT_PLUS_CNT_RST_U0_M  (BIT(0))
+#define PCNT_PLUS_CNT_RST_U0_V  0x1
+#define PCNT_PLUS_CNT_RST_U0_S  0
+
+#define PCNT_DATE_REG          (DR_REG_PCNT_BASE + 0x00fc)
+
+/* PCNT_DATE : R/W ;bitpos:[31:0] ;default: 32'h14122600 ; */
+
+#define PCNT_DATE    0xFFFFFFFF
+#define PCNT_DATE_M  ((PCNT_DATE_V)<<(PCNT_DATE_S))
```

Comment:
```suggestion
#define PCNT_DATE_M  ((PCNT_DATE_V) << (PCNT_DATE_S))
```
if this is not auto-generated

Reply:
Yes, it is autogenerated by regdesc python script. But I wil fix it anyway